js/app/modules/selection/xapian.js

```diff
@@ -76,31 +76,20 @@ const Xapian = new Module.Class({
 
         this._loading = true;
         this.notify('loading');
-        engine.query(query, null, (engine, task) => {
+        engine.query_promise(query).then((results) => {
             this._loading = false;
             this._set_needs_refresh(false);
             this.notify('loading');
+            this._exception = null;
+            if (this._error_state) {
+                this._error_state = false;
+                this.notify('in-error-state');
+            }
 
             let models = [];
             let upper_bound = 0;
-            try {
-                let results = engine.query_finish(task);
-                models = results.models;
-                upper_bound = results.upper_bound;
-                this._exception = null;
-                if (this._error_state) {
-                    this._error_state = false;
-                    this.notify('in-error-state');
-                }
-            } catch (e) {
-                logError(e, 'Failed to load content from engine');
-                this._exception = e;
-                if (!this._error_state) {
-                    this._error_state = true;
-                    this.notify('in-error-state');
-                }
-                return;
-            }
+            models = results.models;
```

Comment:
nitpick: Combine these two lines with the previous two (or even better, `.then({models, upper_bound} => {...})` should work in GJS now)